src/r_bmaps.c

```diff
@@ -23,966 +23,365 @@
+            }
+        }
+    }
+
+    return nobrightmap;
 }
 
-// [crispy] initialize brightmaps
+const byte *R_BrightmapForFlatNum(const int num)
+{
+    if (STRICTMODE(brightmaps))
+    {
+        int i;
+        for (i = 0; i < num_flats_bm; i++)
+        {
+            if (flats_bm[i].num == num)
+            {
+                return flats_bm[i].brightmap->colormask;
+            }
+        }
+    }
+
+    return nobrightmap;
+}
 
-const byte *(*R_BrightmapForTexName) (const char *texname);
-const byte *(*R_BrightmapForSprite) (const int type);
-const byte *(*R_BrightmapForFlatNum) (const int num);
-const byte *(*R_BrightmapForState) (const int state);
+const byte *R_BrightmapForState(const int state)
+{
+    if (STRICTMODE(brightmaps))
+    {
+        int i;
+        for (i = 0; i < num_states_bm; i++)
+        {
+            if (states_bm[i].num == state)
+            {
+                return states_bm[i].brightmap->colormask;
+            }
+        }
+    }
+
+    return nobrightmap;
+}
 
-void R_InitBrightmaps ()
-{
-	if (gamemission == pack_hacx)
-	{
-		bmapflatnum[0] = R_FlatNumForName("FLOOR1_1");
-		bmapflatnum[1] = R_FlatNumForName("FLOOR1_7");
-		bmapflatnum[2] = R_FlatNumForName("FLOOR3_3");
-		bmapflatnum[3] = R_FlatNumForName("NUKAGE1");
-		bmapflatnum[4] = R_FlatNumForName("NUKAGE2");
-		bmapflatnum[5] = R_FlatNumForName("NUKAGE3");
-		bmapflatnum[6] = R_FlatNumForName("BLOOD1");
-		bmapflatnum[7] = R_FlatNumForName("BLOOD2");
-		bmapflatnum[8] = R_FlatNumForName("BLOOD3");
-		bmapflatnum[9] = R_FlatNumForName("SLIME13");
-		bmapflatnum[10] = R_FlatNumForName("SLIME14");
-		bmapflatnum[11] = R_FlatNumForName("SLIME15");
-
-		R_BrightmapForTexName = R_BrightmapForTexName_Hacx;
-		R_BrightmapForSprite = R_BrightmapForSprite_Hacx;
-		R_BrightmapForFlatNum = R_BrightmapForFlatNum_Hacx;
-		R_BrightmapForState = R_BrightmapForState_Hacx;
-	}
-	else
-	if (gamemission == pack_chex)
-	{
-		int lump;
-
-		// [crispy] detect Chex Quest 2
-		lump = W_CheckNumForName("INTERPIC");
-		if (!strcasecmp(W_WadNameForLump(lump), "chex2.wad"))
-		{
-			chex2 = true;
-		}
-
-		R_BrightmapForTexName = R_BrightmapForTexName_Chex;
-		R_BrightmapForSprite = R_BrightmapForSprite_Chex;
-		R_BrightmapForFlatNum = R_BrightmapForFlatNum_None;
-		R_BrightmapForState = R_BrightmapForState_None;
-	}
-	else
-	{
-		// [crispy] only three select brightmapped flats
-		bmapflatnum[0] = R_FlatNumForName("CONS1_1");
-		bmapflatnum[1] = R_FlatNumForName("CONS1_5");
-		bmapflatnum[2] = R_FlatNumForName("CONS1_7");
-
-		R_BrightmapForTexName = R_BrightmapForTexName_Doom;
-		R_BrightmapForSprite = R_BrightmapForSprite_Doom;
-		R_BrightmapForFlatNum = R_BrightmapForFlatNum_Doom;
-		R_BrightmapForState = R_BrightmapForState_Doom;
-	}
+void R_ParseBrightmaps(int lumpnum)
+{
+    u_scanner_t scanner, *s;
+    const char *data = W_CacheLumpNum(lumpnum, PU_CACHE);
+    int length = W_LumpLength(lumpnum);
+
+    if (!num_brightmaps)
+    {
+        brightmap_t brightmap;
```

Comment:
There is already `static const byte nobrightmap[COLORMASK_SIZE] = { 0 }` in line 33. Doesn't that interfere?

Reply:
No, it shouldn't interfere. This is useful if the user wants to overwrite a brightmap in e.g. doom-all, or chex.wad in our case.